test/unit/host_test.rb

```diff
@@ -683,7 +683,21 @@ def teardown
         :domain => domains(:mydomain), :operatingsystem => operatingsystems(:redhat), :subnet => subnets(:two), :puppet_proxy => smart_proxies(:puppetmaster),
         :architecture => architectures(:x86_64), :environment => environments(:production), :managed => true,
         :owner_type => "UserGr(up" # should be Usergroup
-      assert !host.valid?
+      refute host.valid?
+    end
+
+    test "should not save if owner taxonomies do not match host configuration" do
+      org1 = FactoryGirl.create(:organization)
+      loc1 = FactoryGirl.create(:location)
+      user = FactoryGirl.create(:user, :organizations => [], :locations => [])
+      host = FactoryGirl.build(:host, :organization => org1, :location => loc1, :owner => user)
+
+      SETTINGS[:locations_enabled] = true
+      SETTINGS[:organizations_enabled] = true
+      refute host.valid?
+      assert_operator 2, :<=, host.errors[:owner_id].size
+      SETTINGS[:locations_enabled] = false
+      SETTINGS[:organizations_enabled] = false
```

Comment:
in that link you can see it was disabled in before which is the pattern I copied, anyway I'll fix it by keeping original value which is better anyway (will fix the linked test too)